test/SILGen/unreachable_code.swift

```diff
@@ -146,3 +146,10 @@ func testUnreachableCatchClause() {
     print("some error")
   }
 }
+
+func sr13639() -> Int {
+  return Foo.bar
+  struct Foo { // no-warning
+    static var bar = 0
```

Comment:
Can you define a method inside Foo and add a CHECK: line to ensure the method is actually emitted, too?